pip/requirements.txt

```diff
@@ -345,7 +345,7 @@ scandir==1.10.0
 schema==0.7.5
 scikit-build==0.16.7
 scikit-build-core==0.9.3
-scikit-learn==1.2.2
+scikit-learn==1.5.1
```

Comment:
@iarspider , this has to wait as it requires cython 3 (https://github.com/cms-sw/cmsdist/pull/9324#issuecomment-2252838710)